src/operator/tensor/init_op.h

```diff
@@ -731,14 +731,23 @@ inline bool RangeShape(const nnvm::NodeAttrs& attrs,
   return true;
 }
 
+template<typename DType>
 struct linspace_fwd {
-  template<typename DType>
   MSHADOW_XINLINE static void Map(index_t i, double start, double stop, double step,
                                   int req, DType* out) {
     KERNEL_ASSIGN(out[i], req, static_cast<DType>(start + step * i));
   }
 };
 
+// Round towards -infinity for int type
```

Comment:
I think here we can use a macro to generate templated structs for int32 int64 int16 and int8

Reply:
introduced `std::is_integral` for all int types. Thanks for the suggestion!

Reply:
Added `std::is_integral` to cover all int types. Thanks for the suggestion!